### PR TITLE
Implement Eslint Config

### PR DESCRIPTION
**Pull request by erichulburd**
Priorities for this PR:

1. Ensure tests pass. I currently receive following error:
```sh
> BABEL_DISABLE_CACHE=1 babel-node test.js && eslint --fix .

/Users/arbolista/Code/arbolista-spike/node_modules/scroll-js/src/scroll.js:2
import Promise from 'promise';
^^^^^^
SyntaxError: Unexpected token import
    at Object.exports.runInThisContext (vm.js:76:16)
    at Module._compile (module.js:528:28)
    at Module._extensions..js (module.js:565:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/arbolista/Code/arbolista-spike/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/arbolista/Code/arbolista-spike/src/espina/client/router.js:4:1)
```
2. Add necessary rules and fix linting issues - we can discuss which we want to ignore etc.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+		  	"moduleDirectory":[
+		  		"node_modules",
+		  		"src"
+		  	]
+	  	}
+	  }
+  },
+  "rules":{
+  	"react/forbid-prop-types": 0,
+  	"import/no-extraneous-dependencies": ["error", {"devDependencies": ["**/*.test.js", "**/*.spec.js"]}]
```
Comment (DKbyo) [inline .eslintrc:15]: Ignore devDependencies in tests

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "extends": "airbnb",
+  "settings":{
+	  "import/resolver":{
+	  	"node":{
+		  	"moduleDirectory":[
+		  		"node_modules",
+		  		"src"
+		  	]
```
Comment (DKbyo) [inline .eslintrc:9]: Add src as moduleDirectory in order to resolve espina classes in src

File: src/espina/shared/route.js [hunk 3]
```diff
@@ -1,94 +1,88 @@
+/* eslint  import/no-dynamic-require: 0 ,global-require:0 */
 import * as S from 'underscore.string';
```
Comment (DKbyo) [inline src/espina/shared/route.js:2]: Ignore dynamic require

**Review by DKbyo — COMMENTED** [review]
I only ignore two rules: dynamic require and forbid-prop-types. The other rules only require env configuration.

File: .eslintrc (hunk 2 above)
Comment (erichulburd) [reply .eslintrc:9]: 👌

File: package.json [hunk 4]
```diff
@@ -47,22 +48,28 @@
...
-    "react": "^15.3.2",
-    "react-redux": "^4.4.5",
+    "react": "15.3.x",
+    "react-dom": "15.3.x",
+    "react-redux": "4.4.x",
     "redux": "^3.6.0",
     "redux-loop": "^2.2.2",
-    "scroll-js": "^1.2.0",
+    "smooth-scroll": "github:cferdinandi/smooth-scroll",
     "underscore.string": "^3.3.4",
```
Comment (erichulburd) [inline package.json:72]: Is there an issue with npm module?

**Review by erichulburd — COMMENTED** [review]
```
npm run lint -- --fix .
// ✖ 67 problems (59 errors, 8 warnings)

npm run test
module.js:457
    throw err;
    ^

Error: Cannot find module './route_link.component'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/arbolista/Code/arbolista-spike/src/espina/shared/components/route/route_link.test.js:12:1)
    at Module._compile (module.js:556:32)
    at loader (/Users/arbolista/Code/arbolista-spike/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/arbolista/Code/arbolista-spike/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
npm ERR! Test failed.  See above for more details.
```

**Comment by erichulburd** [comment]
When it's ready, take off the don't merge me label.

File: package.json (hunk 4 above)
Comment (DKbyo) [reply package.json:72]: scroll-js is not built, so it can be used in production without build it manually.

Comment (DKbyo) [reply package.json:72]: I only run 

``` bash
eslint --fix test.js src shell generator-tools
```

In order to exclude the files generated by the build.

**Comment by erichulburd** [comment]
6 errores todavia: 

```
> eslint "--fix" "."


/Users/arbolista/Code/arbolista-spike/src/espina/shared/components/route/route_link.test.js
  12:23  error  Unable to resolve path to module './route_link.component'  import/no-unresolved

/Users/arbolista/Code/arbolista-spike/src/espina/shared/test/application_component.test.js
  3:1   error  'espina' should be listed in the project's dependencies. Run 'npm i -S espina' to add it  import/no-extraneous-dependencies
  3:23  error  Unable to resolve path to module 'espina/shared/test/mock/component.mock'                 import/no-unresolved

/Users/arbolista/Code/arbolista-spike/src/espina/shared/test/base_component.test.js
  10:23  error  Unable to resolve path to module './mock/component.mock'  import/no-unresolved

/Users/arbolista/Code/arbolista-spike/src/espina/shared/test/mock/config.mock.js
  1:23  error  'espina' should be listed in the project's dependencies. Run 'npm i -S espina' to add it  import/no-extraneous-dependencies
  1:31  error  Unable to resolve path to module 'espina/shared/test/mock/component.mock'                 import/no-unresolved
```

**Comment by DKbyo** [comment]
I have an error with the gitignore, now eslint works

**Comment by erichulburd** [comment]
All good.

**Pull request merged**